ale_linters/haskell/hls.vim

```diff
@@ -0,0 +1,56 @@
+" Author: Yen3 <yen3rc@gmail.com>
+" Description: A language server for haskell
+"              The file is based on hie.vim (author: Luxed
+"              <devildead13@gmail.com>).  It search more project root files.
+
+call ale#Set('haskell_hls_executable', 'haskell-language-server-wrapper')
+
+function! ale_linters#haskell#hls#FindRootFile(buffer) abort
+    for l:path in ale#path#Upwards(expand('#' . a:buffer . ':p:h'))
+        if filereadable(l:path . 'stack.yaml')
+        \|| filereadable(l:path . 'cabal.project')
+        \|| filereadable(l:path . 'package.yaml')
+        \|| filereadable(l:path . 'hie.yaml')
+            return l:path
+        endif
+    endfor
+
+    return ''
+endfunction
+
+function! ale_linters#haskell#hls#GetProjectRoot(buffer) abort
+    " Search for the stack file first
```

Comment:
s/stack file/project file/ since it searches for many different project files

Reply:
Thanks. Refine the comment.